docs/datascience/data-wrangler-full-documentation.md

```diff
@@ -0,0 +1,199 @@
+| Round                          | Rounds numbers to the specified number of decimal places |
+| Round down (floor)             | Rounds numbers down to the nearest integer |
+| Round up (ceiling)             | Rounds numbers up to the nearest integer |
+| Custom operation               | Automatically create a new column based on examples and the derivation of existing column(s)          |
+
+If there is an operation that is missing that you'd like to see supported in Data Wrangler, please file a feature request in our [Github repo](https://github.com/microsoft/vscode-data-wrangler/issues)
+
+## Modifying previous steps
+
+Each step of the generated code can be modified through the cleaning steps panel. Select the step you want to modify first. Then, as you make changes to the operation (either via code or the operation panel), the effects of your changes on the data will be highlighted in the grid view.
+
+![image](https://github.com/microsoft/vscode-docs/assets/15910920/8d2e20dd-9c2d-4b9c-bdad-3651d1930cc2)
+
+
+## Editing and exporting code
+
+Once you’re done with your data cleaning steps in Data Wrangler, there are 3 ways to export your cleaned dataset from Data Wrangler.
+
+1. **Export code back to Notebook and exit:** This creates a new cell in your Jupyter Notebook with all the data cleaning code you generated packaged up into a clean Python function.
+2. **Export data to a file:** This saves the cleaned dataset as a new CSV or Parquet file onto your machine.
+3. **Copy code to clipboard:** This copies all the code that was generated by Data Wrangler for the data cleaning operations.
+
+
+![image](https://github.com/microsoft/vscode-docs/assets/15910920/798cc251-05a2-4cef-85a1-c99210d48b29)
+
+## Searching for columns
+
+To find a specific column in your dataset, select `Go to column` from the Data Wrangler toolbar and search for the respective column.
+
+![image](https://github.com/microsoft/vscode-docs/assets/15910920/0a8d41ab-d8ba-440b-9e67-312618375989))
+
+## Troubleshooting
+
+### General kernel connectivity issues
+
+For general connectivity issues, please see the "Connecting to a Python kernel" section above on alternative methods to connect. To debug issues related to the local Python interpreter option, one way to potentially fix the issue would be to try installing different versions of the Jupyter and Python extensions. For example, if stable versions of the extensions are installed then one could try installing the pre-release (and vice versa). If the issue is new, then consider reverting to a previous version.
+
+To clear an already cached kernel, you can run the `Data Wrangler: Clear cached runtime` command from the command palette (Ctrl/Cmd+Shift+P).
+
+### UnicodeDecodeError
+
+If you run into a UnicodeDecodeError when opening a data file directly from Data Wrangler, then this could be caused by two possible issues:
```

Comment:
```suggestion
If you run into a `UnicodeDecodeError` when opening a data file directly from Data Wrangler, then this could be caused by two possible issues:
```